docs/examples/user-configuration/user-config-template.yaml

```diff
@@ -1,145 +1,212 @@
-              - containerPort: 11222
-                name: hotrod
-                protocol: TCP
-            readinessProbe:
-              exec:
-                command:
-                  - /bin/bash
-                  - -c
-                  - /opt/datagrid/bin/readinessProbe.sh
-            resources:
-              limits:
-                memory: 1Gi
-            volumeMounts:
-              - mountPath: /opt/datagrid/standalone/configuration/user
-                name: config-volume
-                readOnly: true
-          terminationGracePeriodSeconds: 60
-          volumes:
-            - configMap:
-                name: datagrid-config
+          ports:
+          - containerPort: 8443
+            name: https
+            protocol: TCP
+          - containerPort: 8888
+            name: ping
+            protocol: TCP
+          - containerPort: 11222
+            name: hotrod
+            protocol: TCP
+          readinessProbe:
+            exec:
+              command:
+              - /opt/datagrid/bin/readinessProbe.sh
+            failureThreshold: 5
+            initialDelaySeconds: 17
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 10
+          resources:
+            limits:
+              memory: ${TOTAL_CONTAINER_MEM}Mi
+            requests:
+              cpu: "0.5"
+              memory: ${TOTAL_CONTAINER_MEM}Mi
+          volumeMounts:
+          - mountPath: /opt/datagrid/standalone/data
+            name: srv-data
+          - mountPath: /var/run/secrets/java.io/keystores
+            name: keystore-volume
+          - mountPath: /var/run/secrets/openshift.io/serviceaccount
+            name: service-certs
+          # [user config mod / Add a ConfigMap volume : start]
+          - mountPath: /opt/datagrid/standalone/configuration/user
             name: config-volume
-      triggers:
-        - type: ConfigChange
+            readOnly: true
+          # [user config mod / Add a ConfigMap volume : end]
+        terminationGracePeriodSeconds: 60
+        volumes:
+        - empty-dir: {}
+          name: keystore-volume
+        - name: service-certs
+          secret:
+            secretName: service-certs
+        # [user config mod / Add a ConfigMap : start]
+        - configMap:
+            name: datagrid-config
+          name: config-volume
+        # [user config mod / Add a ConfigMap : end]
+    triggers:
+    - type: ConfigChange
+    volumeClaimTemplates:
+    - metadata:
+        name: srv-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: ${TOTAL_CONTAINER_STORAGE}Gi
 parameters:
-  - description: The name for the application.
-    displayName: Application Name
-    name: APPLICATION_NAME
-    required: true
-    value: datagrid-app
-  - description: Infinispan image.
-    name: IMAGE
-    required: true
-    value: registry.redhat.io/jboss-datagrid-7/datagrid72-openshift
```

Comment:
If you want to keep this as simple as possible, then `APPLICATION_NAME` is the only required param

Reply:
@ryanemerson but wouldn't users need the Data Grid image to create pods with this template?

Reply:
You can just hardcode the default value of the image param, so replace `${IMAGE}` with `registry.redhat.io/jboss-datagrid-7/datagrid72-openshift` and remove the param. But I'm easy either way